In [ ]:
# Ejemplo de Métricas de Regresión
# Importar librerías 

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Lasso
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

In [2]:
# Ruta del dataset
file_path = "C:/Git/vehicles_env/datasets/dataset_facebook_cosmetics_us.csv"

# Cargar dataframe
fb = pd.read_csv(file_path, sep=';')

In [3]:
# divide los datos en características (la matriz X) y una variable objetivo (y)
X = fb.drop('Total Interactions', axis = 1)
y = fb['Total Interactions']

# Dividir los datos de entrenamiento y datos de prueba
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Crear la instancia del objeto estandarizador
scaler = StandardScaler()

# Entrenar el scaler en el conjunto de entrenamiento usando el método fit
scaler.fit(X_train)

# Estandarizar el conjunto de entrenamiento
X_train_st = scaler.transform(X_train)

# Estandarizar el conjunto de prueba
X_test_st = scaler.transform(X_test)

# Definir el modelo 
model = Lasso()

# Entrenar el modelo con los datos de entrenamiento (Estandarizados)
model.fit(X_train_st, y_train)

# Crear predicción para los datos de validación (Estandarizados)
predictions = model.predict(X_test_st)

# Imprimir el valor medio de la variable objetivo en el conjunto de prueba
print('Mean: {:.2f}'.format(y_test.mean()))

# Imprime las métricas principales 
print('Error Absoluto Medio (EAM) : {:.2f}'.format(mean_absolute_error(y_test, predictions))) # Error Absoluto Medio 
print('Error Cuadrático Medio (ECM) : {:.2f}'.format(mean_squared_error(y_test, predictions))) # Error Cuadrático Medio
print('Coeficiente de determinación (R Cuadrado)  : {:.2f}'.format(r2_score(y_test, predictions))) # Coeficiente de determinación (R Cuadrado)

Mean: 188.36
Error Absoluto Medio (EAM) : 35.42
Error Cuadrático Medio (ECM) : 4896.61
Coeficiente de determinación (R Cuadrado)  : 0.93


¡Es correcto!

El error absoluto medio para el conjunto de validación resultó ser alrededor de 35, lo que parece aceptable, considerando que el valor medio de la variable objetivo es 188.36. Pero el error cuadrático medio ha llegado a 5000 (!). Un error tan "intensificado" solo puede ser bueno en comparación con otros modelos.

¡El coeficiente de determinación es 0.93! Quizás recuerdes que incluso el modelo RandomForest() tenía un R² más pequeño, aunque era un algoritmo más complejo. Este es un buen ejemplo de una situación en la que un modelo lineal simple puede funcionar.